### ISSUE TITLE
PMIx Initialization Bug

### ISSUE THREAD

**Issue by rgmiller** [LLNL/UnifyFS#707]
### System information

Type                   | Version/Name
---                    |     ---
Operating System       | RHEL
OS Version             | 8.2
Architecture           | ppc64le
UnifyFS Version        | Latest from Git

### Describe the problem you're observing
When trying to use PMIx, the servers fail to initialize.  The problem is that we're getting the wrong value for the number of servers to expect.  The root cause seems to be using PMIX_UNIV_SIZE for the value.  On Summit at least, this is the number of cores, not the number of servers.  (See `unifyfs_pmix_init()` in [common/src/unifyfs_keyval.c](common/src/unifyfs_keyval.c).)

### Describe how to reproduce the problem
Just build with the "--enable-pmix" config option. 

### Include any warning or errors or releveant debugging data

From `unifyfsd.log.g25n15`:
```
2022-02-09T16:07:07 tid=3102381 @ unifyfs_publish_server_pids() [unifyfs_server_pid.c:151] server initialization timeout
2022-02-09T16:07:07 tid=3102381 @ unifyfs_publish_server_pids() [unifyfs_server_pid.c:166] 4 of 168 servers reported their pids
```

This was from a run with 4 nodes, so all the servers that actually existed did initialize.  It's just that unifyfsd is expecting 168 servers instead of 4.

**Comment by rgmiller** [comment]
A few more details:

Inside `unifyfs_pmix_init()`:
1. The value from the  `PMIx_Get(.. PMIX_UNIV_SIZE...)` function gets assigned to `pmix_univ_nprocs`
2. `pmix_univ_nprocs` is assigned to `kv_nranks`
3. `kv_nranks` is assigned to `glb_pmi_size`

Then, over in `unifyfs_publish_server_pids()`, `glb_pmi_size` is used for the number of servers to expect.

Given the comments over in [this issue](https://github.com/pmix/pmix-standard/issues/121), it seems like we probably want to use `PMIX_JOB_SIZE` instead of `PMIX_UNIV_SIZE`.

**Comment by CamStan** [comment]
Resolved by #712
